### PR TITLE
Noworkergrouplink

### PR DESCRIPTION
**Pull request by kbara**

### CODE REVIEW

**Comment by wingo** [comment]
I think this branch works for Luke because he runs NIC things in the main process and the worker, therefore the main process has already created the group folder when it forks.  I think for us, that is not the case and so somehow the "alias" detects that "group" is already there but it isn't.  This change results in the group dir being created in the worker instead.  That is fine for our use case because we don't actually need to share DMA resources between the main process and the worker, as we're only going to have one worker for the moment and it's the only one that will be doing DMA.  Therefore I think a change like this looks to me to be as correct as anything.

**Comment by kbara** [comment]
I've manually confirmed with snabb top that the worker process is breathing correctly.

Command line tested against:
```./snabb lwaftr run -D 2000  --conf ~/snabb-lwaftr-v3.1.0/src/program/lwaftr/tests/data/icmp_on_fail.conf --reconfigurable --on-a-stick 82:00.1 ```

File: src/core/worker.lua [hunk 1]
```diff
@@ -28,16 +28,23 @@ function start (name, luacode)
    if pid == 0 then
```
Comment (wingo) [inline src/core/worker.lua:28]: Can you please try adding
```
      shm.mkdir(shm.resolve("group/dma"))
```
before line 27 and keeping the shm.alias line uncommented? It seems like a simple fix that would prevent diffs with upstream and the need to comment.

**Review by wingo — CHANGES_REQUESTED** [review]

File: src/core/worker.lua (hunk 1 above)
Comment (kbara) [reply src/core/worker.lua:28]: That's a small variant of what I was working on when asked to PR this solution; I agree it's a much better one.

**Pull request closed without merging**